lib/view_model/node_list/node_create_or_edit_view_model.dart

```diff
@@ -152,4 +153,26 @@ abstract class NodeCreateOrEditViewModelBase with Store {
 
   @action
   void setAsCurrent(Node node) => _settingsStore.nodes[_walletType] = node;
+
+  @action
+  Future<void> scanQRCodeForNewNode() async {
+    String code = await presentQRScanner();
+
+    if (code.isEmpty) {
+      throw Exception('Unexpected scan QR code value: value is empty');
```

Comment:
throwing an exception without being handled will cause the error popup to show instead of showing this error message to the user which is unreasonable.

also, you need to handle the case that the scanned QR code is not empty, but is not valid for this case

I sent you an example of the error file that will come to us if a user scanned a wrong QR code (a QR that does not comply with this code)

Reply:
like wrapping the whole function in a try-catch block

Reply:
Alright,

Fixed it.